network/src/validator_network/network_builder.rs

```diff
@@ -296,30 +303,47 @@ impl NetworkBuilder {
     /// Note: a connectivity manager should only be added if the network is
     /// permissioned.
     pub fn add_connectivity_manager(&mut self) -> &mut Self {
-        let (conn_mgr_reqs_tx, conn_mgr_reqs_rx) = channel::new(
-            self.channel_size,
-            &counters::PENDING_CONNECTIVITY_MANAGER_REQUESTS,
+        let config = ConnectivityManagerBuilderConfig::new(
+            self.network_context().clone(),
+            self.trusted_peers.clone(),
+            self.seed_peers.clone(),
+            self.connectivity_check_interval_ms,
+            2, // Hard-coded constant
+            self.max_connection_delay_ms,
         );
-        self.conn_mgr_reqs_tx = Some(conn_mgr_reqs_tx);
-        let trusted_peers = self.trusted_peers.clone();
-        let seed_peers = self.seed_peers.clone();
-        let max_connection_delay_ms = self.max_connection_delay_ms;
-        let connectivity_check_interval_ms = self.connectivity_check_interval_ms;
-        let pm_conn_mgr_notifs_rx = self.add_connection_event_listener();
-        let conn_mgr = self.executor.enter(|| {
-            ConnectivityManager::new(
-                self.network_context.clone(),
-                trusted_peers,
-                seed_peers,
-                interval(Duration::from_millis(connectivity_check_interval_ms)).fuse(),
+
+        self.connectivity_manager_config = Some(config);
+        self.build_connectivity_manager()
+    }
+
+    /// Build the ConnectivityManager from the provided configuration.
+    fn build_connectivity_manager(&mut self) -> &mut Self {
+        if let Some(config) = self.connectivity_manager_config.take() {
+            let (conn_mgr_reqs_tx, conn_mgr_reqs_rx) = channel::new(
+                self.channel_size,
+                &counters::PENDING_CONNECTIVITY_MANAGER_REQUESTS,
+            );
+            self.conn_mgr_reqs_tx = Some(conn_mgr_reqs_tx);
+            let pm_conn_mgr_notifs_rx = self.add_connection_event_listener();
```

Comment:
There is an obvious question as to whether or not these wires can be included in the config.  There are complexities around the Receiver type which make doing so a non-obvious step.